File: apple/Harness/Services/MockFeatureFlagService.swift
//
//  Copyright 2022 Harness
//

import Foundation

class MockFeatureFlagService: FeatureFlagService {
    func load(_ callback: @escaping()->()) { }
    
    func destroy() { }
    
    func boolVariation(evaluationId: String, _ callback: @escaping(_ result: Bool)->()) {
        callback(true)
    }
    
    func stringVariation(evaluationId: String, _ callback: @escaping(_ result: String)->()) {
        callback("Mocked string response.")
    }
    
    func numberVariation(evaluationId: String, _ callback: @escaping (Int) -> ()) {
        callback(1)
    }
    
    func jsonVariation(evaluationId: String, _ callback: @escaping (String) -> ()) {
        callback("{}")
    }
}

File: android/app/src/main/java/io/harness/services/MockFeatureFlagService.kt
package io.harness.services

import javax.inject.Inject

class MockFeatureFlagService @Inject constructor() : FeatureFlagService {
    override fun load(callback: () -> Unit) = callback()

    override fun destroy(callback: () -> Unit) = callback()

    override fun boolVariation(evaluationId: String, callback: FeatureFlagCallback<Boolean>) = callback(true)

    override fun stringVariation(evaluationId: String, callback: FeatureFlagCallback<String>) = callback("Mocked string reponse.")

    override fun numberVariation(evaluationId: String, callback: FeatureFlagCallback<Number>) = callback(1)

    override fun jsonVariation(evaluationId: String, callback: FeatureFlagCallback<JSONString>) = callback("{}")
}